llvm/lib/Analysis/ScalarEvolution.cpp

```diff
@@ -6858,10 +6858,9 @@ const ConstantRange &ScalarEvolution::getRangeRef(
       ObjectSizeOpts Opts;
       Opts.RoundToAlign = false;
       Opts.NullIsUnknownSize = true;
-      uint64_t ObjSize;
-      if ((isa<GlobalVariable>(V) || isa<AllocaInst>(V) ||
-           isAllocationFn(V, &TLI)) &&
-          getObjectSize(V, ObjSize, DL, &TLI, Opts) && ObjSize > 1) {
+      bool CanBeNull, CanBeFreed;
+      uint64_t ObjSize = V->getPointerDereferenceableBytes(DL, CanBeNull, CanBeFreed);
```

Comment:
ObjSize -> DerefBytes or something. It may not be the full object size.

Reply:
Done